conanfile.py

```diff
@@ -270,6 +278,18 @@ def requirements(self):
             if self._with_xlib_lease:
                 self.requires('libxrandr/1.5.2@bincrafters/stable')
 
+    def system_requirements(self):
+        if tools.os_info.is_linux:
+            pack_names = []
+            if tools.os_info.with_apt and self.options.with_wayland:
+                pack_names += ["libwayland-dev", "libwayland-egl-backend-dev", "wayland-protocols"]
+
+            if pack_names:
+                installer = tools.SystemPackageTool()
+                for item in pack_names:
+                    installer.install(item)
```

Comment:
Do not install, but only check if it's installed: https://github.com/conan-io/conan-center-index/blob/a504b23f1bd435639e92bd063253ed922577395b/recipes/backward-cpp/all/conanfile.py#L87

Reply:
done in last commit 1f4a941

Reply:
well, I don't understand the first time, done in 8fa95c1.
But this is not an usual way to handle missing system package. Currently, conan install missing system package, this simplify CI of package that depend of this one.
Isn't It ?

Reply:
@uilianries now I no longer install system package, many CI build failed ...

[GCC8](https://github.com/bincrafters/conan-mesa/pull/2/checks?check_run_id=745179591#step:5:735)
[GCC9](https://github.com/bincrafters/conan-mesa/pull/2/checks?check_run_id=745179595#step:5:735)
[CLANG6](https://github.com/bincrafters/conan-mesa/pull/2/checks?check_run_id=745179617#step:5:714)
[CLANG7](https://github.com/bincrafters/conan-mesa/pull/2/checks?check_run_id=745179623#step:5:721)
[CLANG8](https://github.com/bincrafters/conan-mesa/pull/2/checks?check_run_id=745179625#step:5:723)
[CLANG9](https://github.com/bincrafters/conan-mesa/pull/2/checks?check_run_id=745179633#step:5:718)

other one is due to a lack of prebuild "m4" package

Reply:
@ledocc The best approach is providing a proxy package or those requirements, you can use this package as reference https://github.com/conan-io/conan-center-index/blob/master/recipes/opengl/all/conanfile.py

Another alternative is installing the packages in CI env, before building. You need to change the CI file.

Reply:
Proxy package looks fun, I'm on it.